CalibTracker/SiStripHitEfficiency/plugins/HitEff.cc

```diff
@@ -592,7 +613,56 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
             }
           }
 
-          if (!tmpTmeas.empty()) {
+          //Test for two consecutive missing hits
+          if (int(TKlayers) - int(prev_TKlayers) == -3) {
+            foundConsMissingHits = true;
+            const DetLayer* detlayer = itm->layer();
+            const LayerMeasurements layerMeasurements{*measurementTrackerHandle, *measurementTrackerEvent};
+            const TrajectoryStateOnSurface tsos = itm->updatedState();
+            std::vector<DetLayer::DetWithState> compatDets = detlayer->compatibleDets(tsos, *thePropagator, *estimator);
+
+            if (misLayer > k_LayersStart && misLayer < k_LayersAtTOBEnd && previousMisLayer > k_LayersStart &&
+                previousMisLayer < k_LayersAtTOBEnd) {  //Barrel case
```

Comment:
Again, why not
```suggestion
            if (misLayer > k_LayersStart && misLayer <= k_LayersAtTOBEnd && previousMisLayer > k_LayersStart &&
                previousMisLayer <= k_LayersAtTOBEnd) {  //Barrel case
```

Reply:
I implemented this as it was indeed an error on my side, thank you !